frontend/server/handlers/artifacts.ts

```diff
@@ -204,7 +221,7 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }) {
                 .toString()
                 .trim() + '\n';
             if (i === matchingFiles.length - 1) {
-              res.send(contents);
+              res.send(peek > 0 ? contents.slice(0, peek) : contents);
```

Comment:
As I mentioned in https://github.com/kubeflow/pipelines/pull/2992#issuecomment-598085681, I'd prefer a proper peek behavior here as well, to avoid scalability issues.

Reply:
Ok. Might need to research more on gcs sdk, cuz I am not terribly familiar with it. But shld not be an issue.